src/opt/MangleOpenCLBuiltins.cpp

```diff
@@ -18,7 +18,8 @@ static std::string mangleMathOCLBuiltin(const llvm::Function &func) {
     std::string mangledName;
     llvm::Type *retType = func.getReturnType();
     // spirv OpenCL builtins are used for double types only
-    Assert(retType->isVectorTy() && llvm::dyn_cast<llvm::FixedVectorType>(retType)->getElementType()->isDoubleTy() ||
+    Assert(retType->isVectorTy() && (llvm::isa<llvm::FixedVectorType>(retType) &&
+                                     llvm::dyn_cast<llvm::FixedVectorType>(retType)->getElementType()->isDoubleTy()) ||
```

Comment:
Unfortunately, `isa` does is not understood by the static analyzer to mean that `dyn_cast` will not return `nullptr`. An explicit check for `nullptr` is required here.
```suggestion
#ifdef debug
    Assert(llvm::isa<llvm::FixedVectorType>(retType));
    const auto fvt = llvm::dyn_cast<llvm::FixedVectorType>(retType);
    Assert(retType->isVectorTy() && (fvt != nullptr && fvt->getElementType()->isDoubleTy()) ||
// ...
#endif
```
The additional assert for `isa` is optional here. `#ifdef DEBUG` is used to guard the stack allocation of fvt. Without this guard, the allocation will also be done on a `Release` build, where `fvt` remains unused and will issue a warning/error.